dipy/tracking/tracker_deterministic.pyx

```diff
@@ -0,0 +1,70 @@
+# cython: boundscheck=False
+# cython: initializedcheck=False
+# cython: wraparound=False
+# cython: Nonecheck=False
+
+cimport cython
+from cython.parallel import prange
+import numpy as np
+cimport numpy as cnp
+
+from dipy.core.interpolation cimport trilinear_interpolate4d_c
+from dipy.direction.pmf cimport PmfGen
+from dipy.tracking.stopping_criterion cimport StoppingCriterion
+from dipy.utils.fast_numpy cimport (copy_point, cumsum, norm, normalize,
+                                    where_to_insert, random)
+from dipy.tracking.fast_tracking cimport get_pmf, TrackerParameters
+from libc.stdlib cimport malloc, free
+
+
+cdef int deterministic_tracker(double* point,
+                               double* direction,
+                               TrackerParameters params,
+                               double* stream_data,
+                               PmfGen pmf_gen) noexcept nogil:
+    cdef:
+        cnp.npy_intp i, max_idx
+        double max_value=0
+        double* newdir
+        double* pmf
+        double cos_sim
+        cnp.npy_intp len_pmf=pmf_gen.pmf.shape[0]
+
+    if norm(direction) == 0:
+        return 1
```

Comment:
Similar to the comment about the `return 0` statements, should these be declared as a constant and be used across the code?

Reply:
We have many cython functions with 0/1+ status returns, e.g. `dipy.core.interpolation.trilinear_interpolate4d_c`.

I agree constant could clarify the code. @skoudoro what do you suggest?